pints/tests/test_slice_stepout.py

```diff
@@ -0,0 +1,350 @@
+        self.assertEqual(mcmc._l, None)
+        self.assertEqual(mcmc._r, None)
+        self.assertEqual(mcmc._temp_l, None)
+        self.assertEqual(mcmc._temp_r, None)
+        self.assertEqual(mcmc._k, None)
+        self.assertEqual(mcmc._j, None)
+        self.assertEqual(mcmc._fx_l, None)
+        self.assertEqual(mcmc._fx_r, None)
+
+        self.assertEqual(mcmc._m, 50)
+        self.assertEqual(mcmc._active_param_index, 0)
+
+    def test_first_run(self):
+        """
+        Tests the very first run of the sampler.
+        """
+        # Create log pdf
+        log_pdf = pints.toy.GaussianLogPDF([2, 4], [[1, 0], [0, 3]])
+
+        # Create mcmc
+        x0 = np.array([2., 4.])
+        mcmc = pints.SliceStepoutMCMC(x0)
+
+        # Ask should fail if _ready_for_tell flag is True
+        with self.assertRaises(RuntimeError):
+            mcmc._ready_for_tell = True
+            mcmc.ask()
+
+        # Undo changes
+        mcmc._ready_for_tell = False
+
+        # Check whether _running flag becomes True when ask() is called
+        # Check whether first iteration of ask() returns x0
+        self.assertFalse(mcmc._running)
+        self.assertTrue(np.all(mcmc.ask() == x0))
+        self.assertTrue(mcmc._running)
+        self.assertTrue(mcmc._ready_for_tell)
+
+        # Tell should fail when log pdf of x0 is infinite
+        with self.assertRaises(ValueError):
+            fx = np.inf
+            mcmc.tell(fx)
+
+        # Calculate log pdf for x0
+        fx = log_pdf.evaluateS1(x0)[0]
+
+        # Tell should fail when _ready_for_tell is False
+        with self.assertRaises(RuntimeError):
+            mcmc._ready_for_tell = False
+            mcmc.tell(fx)
+
+        # Undo changes
+        mcmc._ready_for_tell = True
+
+        # Test first iteration of tell(). The first point in the chain
+        # should be x0
+        self.assertTrue(np.all(mcmc.tell(fx) == x0))
+
+        # We update the current sample
+        self.assertTrue(np.all(mcmc._current == x0))
+        self.assertTrue(np.all(mcmc._current == mcmc._proposed))
+
+        # We update the _current_log_pdf value used to generate the new slice
+        self.assertEqual(mcmc._current_log_pdf, fx)
+
+        # Check that the new slice has been constructed appropriately
+        self.assertTrue(mcmc._current_log_y < mcmc._current_log_pdf)
+
+        # Check flag
+        self.assertTrue(mcmc._first_expansion)
+
+    def test_run(self):
+        """
+        Test multiple MCMC iterations of the sampler on a
+        Multivariate Gaussian.
+        """
+        # Set seed for monitoring
+        np.random.seed(2)
+
+        # Create log pdf
+        log_pdf = pints.toy.GaussianLogPDF([2, 4], [[1, 0], [0, 3]])
+
+        # Create mcmc
+        x0 = np.array([1, 1])
+        mcmc = pints.SliceStepoutMCMC(x0)
+
+        mcmc.set_w(30)
+        mcmc.set_a(2)
+
+        # Run multiple iterations of the sampler
+        chain = []
+        while len(chain) < 300:
```

Comment:
Why 300? Please make this number as low as possible without ruining the test. Remember everyone needs to run this test every time they change a bit of PINTS code